pkg/controller/cloud/node_lifecycle_controller_test.go

```diff
@@ -258,10 +286,20 @@ func Test_NodesDeleted(t *testing.T) {
 			eventBroadcaster.StartLogging(klog.Infof)
 			cloudNodeLifecycleController.MonitorNodes()
 
-			if !reflect.DeepEqual(testcase.fnh.DeletedNodes, testcase.deleteNodes) {
-				t.Logf("actual nodes: %v", testcase.fnh.DeletedNodes)
-				t.Logf("expected nodes: %v", testcase.deleteNodes)
-				t.Error("unexpected deleted nodes")
+			updatedNode, err := clientset.CoreV1().Nodes().Get(context.TODO(), testcase.existingNode.Name, metav1.GetOptions{})
+			if err != nil {
+				if testcase.expectedNode != nil {
+					t.Fatalf("error getting updated nodes: %v", err)
+				} else {
+					if !strings.Contains(err.Error(), "not found") {
```

Comment:
We should use [IsNotFound()](https://github.com/kubernetes/apimachinery/blob/7e441e0f246a2db6cf1855e4110892d1623a80cf/pkg/api/errors/errors.go#L480)